src/main/java/pl/cleankod/exchange/entrypoint/ExceptionHandlerAdvice.java

```diff
@@ -5,15 +5,36 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import pl.cleankod.exchange.core.usecase.CurrencyConversionException;
 import pl.cleankod.exchange.entrypoint.model.ApiError;
+import pl.cleankod.exchange.provider.nbp.exceptions.NbpAPIException;
 
 @ControllerAdvice
 public class ExceptionHandlerAdvice {
 
     @ExceptionHandler({
             CurrencyConversionException.class,
-            IllegalArgumentException.class
     })
     protected ResponseEntity<ApiError> handleBadRequest(CurrencyConversionException ex) {
         return ResponseEntity.badRequest().body(new ApiError(ex.getMessage()));
     }
+
+    @ExceptionHandler({
+            IllegalArgumentException.class
+    })
+    protected ResponseEntity<ApiError> handleIllegalArgumentExcheption(IllegalArgumentException ex) {
+        return ResponseEntity.badRequest().body(new ApiError("Please check the provided arguments"));
+    }
+
+    @ExceptionHandler({
+            NbpAPIException.class,
+    })
+    protected ResponseEntity<ApiError> handleNbpAPIException(NbpAPIException ex) {
+        return ResponseEntity.badRequest().body(new ApiError(ex.getMessage()));
```

Comment:
I don't think that bad request is the good HTTP response code here. Which one do you think is more suitable?